### PR TITLE
Created manual valgrind tests, fixed few memory leaks

### PR DESCRIPTION
**Pull request by kravemir**

### CODE REVIEW

File: src/lib/ddcci.c [hunk 1]
```diff
@@ -361,6 +362,8 @@ static int i2c_read(struct monitor* mon, unsigned int addr, unsigned char *buf,
 		struct i2c_msg             i2cmsg;
 		int i;
 	
+		memset(&msg_rdwr, 0, sizeof(msg_rdwr));
```
Comment (drinkcat) [inline src/lib/ddcci.c]: I'd just do:
```
struct i2c_rdwr_ioctl_data msg_rdwr = { 0 };
```

File: src/lib/ddcci.c [hunk 2]
```diff
@@ -1100,6 +1103,10 @@ int ddcci_close(struct monitor* mon)
 			free(mon->caps.vcp[i]);
 		}
 	}
+
+	if(mon->caps.raw_caps) {
+		free(mon->caps.raw_caps);
+	}
```
Comment (drinkcat) [inline src/lib/ddcci.c]: free does not mind NULL parameter, so you don't need the if block.

File: scripts/build_gnome_supp.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (drinkcat) [inline scripts/build_gnome_supp.sh]: I think `$(dirname "$0")` should do the same.

File: scripts/build_gnome_supp.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
```
Comment (drinkcat) [inline scripts/build_gnome_supp.sh]: Don't forget copyright headers in these files too.

File: src/lib/ddcci.c (hunk 1 above)
Comment (kravemir) [reply src/lib/ddcci.c]: Does `... = { 0 }` set whole struct to zero?

File: src/lib/ddcci.c (hunk 2 above)
Comment (kravemir) [reply src/lib/ddcci.c]: True. Thanks for reviewing, feel free to do complete review.

I'll be leaving to work soon, and today evening I'm travelling on a short vacation. I'll be back in couple of days, then I'll correct points mentioned in review.

I intentionally left PR open, wondering whether, or hoping, that someone would do a review.

File: scripts/build_gnome_supp.sh (hunk 3 above)
Comment (kravemir) [reply scripts/build_gnome_supp.sh]: I took it from this: https://stackoverflow.com/a/246128/409102. I guess, there was a reason, why it's such complex. I think it's something with symlinks, and.. I don't know,.. some magic?

File: src/lib/ddcci.c (hunk 1 above)
Comment (drinkcat) [reply src/lib/ddcci.c]: Yep (https://stackoverflow.com/questions/11152160/initializing-a-struct-to-0?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa has a good reference too C99 standard).

File: scripts/build_gnome_supp.sh (hunk 3 above)
Comment (drinkcat) [reply scripts/build_gnome_supp.sh]: You could feed the output through `readlink -f`, but I don't think this is really necessary: you don't really care about the absolute path here?

File: scripts/build_gnome_supp.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ ! -d ${DIR}/tmp/GNOME.supp ]; then
```
Comment (drinkcat) [inline scripts/build_gnome_supp.sh]: Quotes "${DIR}/tmp/GNOME.supp" (and everywhere below where you use ${DIR}

File: scripts/build_gnome_supp.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+    mkdir -p ${DIR}/tmp;
```
Comment (drinkcat) [inline scripts/build_gnome_supp.sh]: No ; needed.

File: scripts/common_test.sh [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
+[ "$EUID" -ne 0 ] && echo "Run as root" && exit 1
```
Comment (drinkcat) [inline scripts/common_test.sh]: I like if's better, but, really, up to you.

Comment (kravemir) [reply scripts/common_test.sh]: Makes sense, more readable.

File: scripts/build_gnome_supp.sh (hunk 6 above)
Comment (kravemir) [reply scripts/build_gnome_supp.sh]: Actually, not worth change itself. But, there are missing `"...${DIR}..."`.

File: scripts/build_gnome_supp.sh (hunk 4 above)
Comment (kravemir) [reply scripts/build_gnome_supp.sh]: I'm not that much into `law`/`copyright` stuff,... Why does it matter?

Comment (kravemir) [reply scripts/build_gnome_supp.sh]: Funny way to make `+126` changes: https://github.com/ddccontrol/ddccontrol/pull/55/commits/e84074cc1c7abb9b0d8ca9006ade8dd2893c89d2

Comment (StefanBruens) [reply scripts/build_gnome_supp.sh]: You should not add the full boilerplate on each file, but just add an SPDX header.

Comment (kravemir) [reply scripts/build_gnome_supp.sh]: What's SPDX header?

Comment (StefanBruens) [reply scripts/build_gnome_supp.sh]: First matches when searching for "SPDX header"
https://lwn.net/Articles/739183/
https://github.com/david-a-wheeler/spdx-tutorial/blob/master/README.md

Comment (kravemir) [reply scripts/build_gnome_supp.sh]: @StefanBruens So, I would be:

```
# Copyright(c) 2018 Miroslav Kravec (kravec.miroslav@gmail.com)
#
# SPDX-License-Identifier: GPL-2.0
```

Correct?

Comment (kravemir) [reply scripts/build_gnome_supp.sh]: @StefanBruens I found it,.. way too long `README`: https://github.com/david-a-wheeler/spdx-tutorial/blob/master/README.md#spdx-license-expressions-in-source-code-files

Comment (kravemir) [reply scripts/build_gnome_supp.sh]: @StefanBruens anyway, thanks for info.

**Comment by kravemir** [comment]
@drinkcat thanks for review. I've incorporated changed based on your comments into PR. I didn't change minor code-style issues.

If you have a spare time, you could do a review of whole unreleaded changes: https://github.com/ddccontrol/ddccontrol/compare/0.4.4...master

**Comment by drinkcat** [comment]
@kravemir Thanks for the fixes! I did a semi-random pass at older commits and added a few reviews.

**Pull request merged**